pkg/topology/mock/mock.go

```diff
@@ -128,11 +138,27 @@ func (*mock) NeighborhoodDepth() uint8 {
 	return 0
 }
 
+func (m *mock) IsWithinDepth(adr swarm.Address) bool {
+	return false
+}
+
+func (m *mock) EachNeighbor(f topology.EachPeerFunc) error {
+	return m.EachPeer(f)
+}
+
+func (*mock) EachNeighborRev(topology.EachPeerFunc) error {
```

Comment:
where do we need this function?

Reply:
we'll remove that in a future PR since anatol is also working on this code :+1: